### DISCUSSION TITLE
Assign SVG a Color Profile and Convert to Another Format Using That Profile

### DISCUSSION THREAD

**Discussion by mackerbruny** [ImageMagick/ImageMagick#4703] [Help] [answered]
I believe it is possible to [specify a CMYK color profile](https://stackoverflow.com/questions/3405689/svg-image-with-cmyk-colours-is-it-possible) and color mix within the SVG format. Most of what I read seems to indicate that doing so produces black fills. 

I'm wondering if it is possible with ImageMagick to input an SVG with color profile and mix applied and then convert it to another file format, like PDF, with the color profile and mix from the SVG.

I found [one other post](https://github.com/ImageMagick/ImageMagick/discussions/2902) pertaining to this topic.

**Comment by snibgo** [comment]
IM delegates rasterization of SVG to MSVG, RSVG or Inkscape. Of these, Inkscape is the most complete.

So your question boils down to: Does Inkscape support CMYK profiles, and can it generate rasters with that profile? That would be a question for an Inkscape forum. If it does, then I expect IM would support that, or could be changed to support that.

The question in the other post says "Unfortunately, Inkscape has no support for writing exported files with such a color profile."

**Reply by mackerbruny** [reply]
Thank you. Unfortunately, I'm very out of my depth here. I don't understand how Inkscape is tied up with IM. I would want to use IM in a web application. I would want a user to be able to import SVG and get JPG PNG EPS and PDF out of it. The PDFs would need to be in RGB and CMYK. The EPS would need to be in CMYK. The JPG would need to be in RGB and CMYK.

In my limited understanding, I presumed that an image converter like IM could take color data from either the SVG itself or metadata and export files with the appropriate colors and color profiles.

Would this be easier using PDF as the input file instead of SVG?

**Comment by snibgo** [comment]
In what sense do you mean "easier"? To read an SVG and write a JPG, the command could be:
```
magick in.svg out.jpg
```
To create multiple outputs, it might be:
```
magick in.svg +write out.jpg +write out.png out.pdf
```
If Inkscape writes the SVG with an embedded CMYK profile, then converting that to sRGB (if that is what you want) is slightly more complex.

Do you have a sample SVG file? Explaining is easier with a real example. Also: what version of IM do you have, on what platform?

**Reply by mackerbruny** [reply]
I am not a developer. I do not use IM. I'm asking these questions as initial research for a product I'd like to build. I would be building a web app where the user can upload an SVG (or PDF). The web app would then be able to create several color variations of that SVG (or PDF) and then export RGB and CMYK files (RGB for JPG PNG SVG CMYK for JPG PDF EPS) of those color variations.

I'd like to achieve this without creating a proprietary file format, and SVG seems like the most robust format for modifying shapes and colors as a vector on the web.

I appreciate your patience and knowledge. Thanks!

**Comment by snibgo** [comment]
As you may know, IM is a raster image processor. If your user uploads a vector file, which might be SVG or PDF, IM won't edit vector elements and create an output with vector data in a SVG or PDF.

Instead, IM will rasterize the vector data, ie make an image with pixels. It can then write that raster image to SVG or PDF.

**Reply by mackerbruny** [reply]
Thank you for that information. I did not know that. Are there vector image processors out there? I can't imagine why IM couldn't export an SVG with vector information if it had an SVG to start from. An SVG that only contains a raster image defeats the purpose of the SVG format (scalable VECTOR graphic). I can better understand a PDF that only contains a raster image, but that is also less than desirable most of the time.

**Comment by snibgo** [comment]
> I can't imagine why IM couldn't export an SVG with vector information...

Because IM in a raster image processor, not a vector image processor.

**Reply by mackerbruny** [reply]
Alright, so I'd have to use something else to export SVGs PDFs and EPSs since my purposes require vector info in those formats. I'm still not sure if IM could export a CMYK JPG from an SVG. Could IM export a CMYK JPG from an RGB PDF?

Do you know of vector image processors that are frameworks or libraries? I don't know what you would call IM, but I'm not looking for a tool like Illustrator or Inkscape, etc. I want to build a tool that processes vector and raster images, not use one.

**Comment by snibgo** [comment, marked as answer]
> I'm still not sure if IM could export a CMYK JPG from an SVG.

From a sRGB SVG, yes. I don't know if an SVG can be CMYK.

> Could IM export a CMYK JPG from an RGB PDF?

Yes.

> Do you know of vector image processors that are frameworks or libraries?

An SVG is a text file. It is fairly easy to parse into a data structure, and there are plenty of tools that can do that. Then elements can be edited, and an SVG can be re-created.

For very simple editing, the SVG file can be edited directly, either by hand in a text editor or with tools such as sed.

**Reply by mackerbruny** [reply]
Thank you. I want to mark something as the answer. Which response of yours would you like marked? This one?

**Comment by snibgo** [comment]
Any. It doesn't matter.
